### PR TITLE
Bail on sharing comments with siblings if key is a string

### LINKED ISSUE #4995: Crash on transform ("To get a node path the parent needs to exist") when using a comment after a label before a for..of loop for a non-array iterable
This is using only the `es2015` transform on the latest babel (6.18.2).

This code fails:
```js
myLabel: //woops
for (let a of {}) {
  continue myLabel;
}
```

Error: `Invariant Violation: To get a node path the parent needs to exist`

This code works fine (without comment):
```js
myLabel:
for (let a of {}) {
  continue myLabel;
}
```

Also OK (`for...in` instead of `for...of`)
```js
myLabel: //woops
for (let a of {}) {
  continue myLabel;
}
```

Also fine (loop over array literal):
```js
myLabel: //woops
for (let a of []) {
  continue myLabel;
}
```

Yay edge-cases!

### PR DESCRIPTION
**Pull request by existentialism**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | Y
| Major: Breaking Change?  | N
| Minor: New Feature?      | N
| Deprecations?            | N
| Spec Compliancy?         | N
| Tests Added/Pass?        | Y/Y
| Fixed Tickets            | Fixes #4995
| License                  | MIT
| Doc PR                   | 
| Dependency Changes       |

### CODE REVIEW

File: packages/babel-plugin-transform-es2015-for-of/test/fixtures/regression/label-object-with-comment-4995/actual.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+myLabel: //woops
+for (let a of {}) {
```
Comment (hzoo) [inline packages/babel-plugin-transform-es2015-for-of/test/fixtures/regression/label-object-with-comment-4995/actual.js]: `{}` isn't valid anyway so it should be anything else like b

Comment (hzoo) [inline packages/babel-plugin-transform-es2015-for-of/test/fixtures/regression/label-object-with-comment-4995/actual.js]: If we wanted to test validity then could make an exec.js test

Comment (xtuc) [inline packages/babel-plugin-transform-es2015-for-of/test/fixtures/regression/label-object-with-comment-4995/actual.js]:
```javascript
> for (let a of {}) {}
TypeError: (intermediate value)[Symbol.iterator] is not a function
    at repl:1:15
    at sigintHandlersWrap (vm.js:22:35)
    at sigintHandlersWrap (vm.js:96:12)
    at ContextifyScript.Script.runInThisContext (vm.js:21:12)
    at REPLServer.defaultEval (repl.js:313:29)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.<anonymous> (repl.js:513:10)
    at emitOne (events.js:101:20)
    at REPLServer.emit (events.js:188:7)

```

File: packages/babel-traverse/src/path/comments.js [hunk 2]
```diff
@@ -5,6 +5,9 @@
  */
 
 export function shareCommentsWithSiblings() {
+  // NOTE: this assumes numbered keys
+  if (typeof this.key === "string") return;
```
Comment (Kovensky) [inline packages/babel-traverse/src/path/comments.js]: I wonder what value does the `key` get in a `LabeledStatement`... or does the sharing happen from the `ForOfStatement` side? That would be bad 😆

Comment (hzoo) [inline packages/babel-traverse/src/path/comments.js]: the key is basically the position in `body` or a `block` via an integer but if it's part of another node it's just the property name of the key (so for for-of it's like left, right, body). For a `LabeledStatement` its an integer too.

**Pull request merged**